File: public/examples/integrations/mcp-servers/clickup_api/update_goal_details_example_call_tool.py
import json
from arcadepy import Arcade

client = Arcade()  # Automatically finds the `ARCADE_API_KEY` env variable

USER_ID = "{arcade_user_id}"
TOOL_NAME = "ClickupApi.UpdateGoalDetails"

auth_response = client.tools.authorize(
    tool_name=TOOL_NAME,
    user_id=USER_ID,
)

if auth_response.status != "completed":
    print(f"Click this link to authorize: {auth_response.url}")

# Wait for the authorization to complete
client.auth.wait_for_completion(auth_response)

tool_input = {
    'goal_color': '#FF5733',
    'goal_description': 'Increase team productivity by 20%',
    'goal_due_date': 1699996800,
    'goal_id': '123e4567-e89b-12d3-a456-426614174000',
    'goal_name': 'Productivity Boost',
    'new_owners_to_add': ['user_1', 'user_2'],
    'remove_owners_user_ids': ['user_3']
}

response = client.tools.execute(
    tool_name=TOOL_NAME,
    input=tool_input,
    user_id=USER_ID,
)
print(json.dumps(response.output.value, indent=2))

File: public/examples/integrations/mcp-servers/clickup_api/update_goal_details_example_call_tool.js
import { Arcade } from "@arcadeai/arcadejs";

const client = new Arcade(); // Automatically finds the `ARCADE_API_KEY` env variable

const USER_ID = "{arcade_user_id}";
const TOOL_NAME = "ClickupApi.UpdateGoalDetails";

// Start the authorization process
const authResponse = await client.tools.authorize({
  tool_name: TOOL_NAME,
  user_id: USER_ID,
});

if (authResponse.status !== "completed") {
  console.log(`Click this link to authorize: ${authResponse.url}`);
}

// Wait for the authorization to complete
await client.auth.waitForCompletion(authResponse);

const toolInput = {
  "goal_color": "#FF5733",
  "goal_description": "Increase team productivity by 20%",
  "goal_due_date": 1699996800,
  "goal_id": "123e4567-e89b-12d3-a456-426614174000",
  "goal_name": "Productivity Boost",
  "new_owners_to_add": [
    "user_1",
    "user_2"
  ],
  "remove_owners_user_ids": [
    "user_3"
  ]
};

const response = await client.tools.execute({
  tool_name: TOOL_NAME,
  input: toolInput,
  user_id: USER_ID,
});

console.log(JSON.stringify(response.output.value, null, 2));
